creation/lib/cvWCreate.py

```diff
@@ -78,11 +78,68 @@ def create_client_mapfile(mapfile_fname, my_DN, factory_DNs, schedd_DNs, collect
     return
 
 
+def find_multilines(config_text):
+    """
+    Parses condor config file looking for multiline entries
+    Args:
+        config_text: string, contents of a condor configuration file
+    Returns:
+        multi: dictionary. keys are first line of multi line config
+                         values are the rest of the multi line config
+                         keeping original formatting
+    """
+    multi = {}
+    tag = None
+    dict_key = None
+    for line in config_text:
+        parts = line.split()
+        if not tag:
+            for idx in range(len(parts)):
+                if parts[idx].startswith("@="):
+                    tag = parts[idx].replace("=", "").strip()
+                    dict_key = parts[idx - 1].strip()
+                    multi[dict_key] = "".join(parts[idx:]) + "\n"
+        else:
+            val = multi[dict_key]
+            val += line
+            multi[dict_key] = val
+            for idx in range(len(parts)):
+                if tag in parts[idx]:
+                    tag = False
+    return multi
+
+
+def parse_configs_for_multis(conf_list):
+    """
+    processes list of condor config files searching for multi line configurations
+    Args:
+       conf_list: string, output of condor_config_val -config
+    Returns:
+        multi: dictionary. keys are first line of multi line config
+                         values are the rest of the multi line config
+                         keeping original formatting
+    """
+    multi = {}
+    for conf in conf_list:
+        conf = conf.strip()
+        if os.path.exists(conf):
+            with open(conf) as fd:
+                text = fd.readlines()
+                pdict = find_multilines(text)
+            multi.update(pdict)
+    return multi
+
+
 #########################################
 # Create frontend-specific condor_config
 def create_client_condor_config(config_fname, mapfile_fname, collector_nodes, classad_proxy):
```

Comment:
How is condor_config_val -dump outputting the multiline values? 
If it is maintaining the tag notation, this could be used to parse the values including multiline.
Otherwise, there is the potential problem of a macro/attribute being multiline in one file and single line in another. We would pick always the multiline value which could not be the last one.
To do a good job we'd have to parse all attributes file by file ... 
And there is also the complication of the substitutions that could happen also in the multiline values. I'm not sure if condor_config_val does that as well. Or if we don't need to worry about it.